### PR TITLE
fix: move `<MakeswiftProvider>` to the root layout + Next 15 compat

### PR DESCRIPTION
**Pull request by agurtovoy**
## What/Why?

Makeswift [integration setup](https://docs.makeswift.com/developer/app-router/installation#geting-started) + Next 15 compatibility fixes.

### CODE REVIEW

File: core/app/[locale]/(default)/[...rest]/page.tsx [hunk 1]
```diff
@@ -20,27 +19,24 @@ export async function generateStaticParams() {
...
         locale: locale === defaultLocale ? undefined : locale,
       })),
     );
 }
 
-export default async function CatchAllPage({ params }: { params: CatchAllParams }) {
-  const path = `/${params.rest.join('/')}`;
+export default async function CatchAllPage({ params }: { params: Promise<CatchAllParams> }) {
+  const { locale, rest } = await params;
+  const path = `/${rest.join('/')}`;
```
Comment (agurtovoy) [inline core/app/[locale]/(default)/[...rest]/page.tsx:30]: Next.js 15 compat

File: core/app/[locale]/(default)/[...rest]/page.tsx [hunk 2]
```diff
@@ -20,27 +19,24 @@ export async function generateStaticParams() {
...
+    locale: locale === defaultLocale ? undefined : locale,
   });
 
   if (snapshot == null) return notFound();
 
-  return (
-    <MakeswiftProvider>
-      <MakeswiftPage snapshot={snapshot} />
-    </MakeswiftProvider>
-  );
```
Comment (agurtovoy) [inline core/app/[locale]/(default)/[...rest]/page.tsx:43]: Move `<MakeswiftProvider>` to the root layout to match our recommended setup.

File: core/app/[locale]/layout.tsx [hunk 3]
```diff
@@ -12,10 +12,13 @@ import '../globals.css';
 import { client } from '~/client';
 import { graphql } from '~/client/graphql';
 import { revalidate } from '~/client/revalidate-target';
+import { MakeswiftProvider } from '~/lib/makeswift/provider';
 
 import { Notifications } from '../notifications';
 import { Providers } from '../providers';
 
+import '~/lib/makeswift/components';
```
Comment (agurtovoy) [inline core/app/[locale]/layout.tsx:20]: Import components registered w/ Makeswift in the root layout.

File: core/components/ui/carousel/carousel.makeswift.client.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
+'use client';
```
Comment (agurtovoy) [inline core/components/ui/carousel/carousel.makeswift.client.tsx:1]: Separate the Makeswift client component implementation from the component registration; the registration needs to happen both on the client and on the server.

**Review by chanceaclark — APPROVED** [review]

**Review by migueloller — APPROVED** [review]

**Pull request merged**